src/languages/r.js

```diff
@@ -2,59 +2,105 @@
 Language: R
 Description: R is a free software environment for statistical computing and graphics.
 Author: Joe Cheng <joe@rstudio.org>
+Contributors: Konrad Rudolph <konrad.rudolph@gmail.com>
 Website: https://www.r-project.org
 Category: scientific
 */
 
 export default function(hljs) {
-  var IDENT_RE = '([a-zA-Z]|\\.[a-zA-Z.])[a-zA-Z0-9._]*';
+  // FIXME: Support Unicode identifiers.
+  var IDENT_RE = '(([a-zA-Z]|\\.[._a-zA-Z])[._a-zA-Z0-9]*)|\\.(!=\\d)'
+
+  var ROXYGEN = {
+    className: 'docTag',
+    begin: "#'",
+    end: '$',
+    contains: [
+      {
+        className: 'meta',
+        begin: '@[a-zA-Z]+',
+        relevance: 0
+      },
+      {
+        className: 'built_in',
+        begin: '\\\\[a-zA-Z]+',
+        relevance: 0
+      }
+    ]
+  }
 
   return {
     name: 'R',
     contains: [
-      hljs.HASH_COMMENT_MODE,
+      {
+        className: 'string',
+        contains: [hljs.BACKSLASH_ESCAPE],
+        variants: [
+          hljs.END_SAME_AS_BEGIN({ begin: /r"(-*)\(/, end: /\)(-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r"(-*)\{/, end: /\}(-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r"(-*)\[/, end: /\](-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r'(-*)\(/, end: /\)(-*)'/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r'(-*)\{/, end: /\}(-*)'/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r'(-*)\[/, end: /\](-*)'/ }),
+          {begin: '"', end: '"', relevance: 0},
+          {begin: "'", end: "'", relevance: 0}
+        ],
+        relevance: 10
```

Comment:
Don't several different languages have `r` strings now?  Not sure relevance 10 should apply.  10 really means "this is pretty much exclusive to our language".

Reply:
The syntax for these raw strings is different in Python. That said, C++’s syntax (while still different) overlaps with R’s. Ideally I’d like setting the relevance “high, but not exclusive” but the documentation says

> When setting an explicit value it’s recommended to use either 10 or 0.

I can skip this and use the default if that’s more appropriate.

Reply:
I think that's there only to avoid people using entirely random values... we probably could add some further clarification (esp. for languages in core).

Grrr, it seems Python is flagging it's own r|ur|u strings as 10 as well... I may need to tone that down... in general bumping relevance for someone as common as a string is usually more likely to get you into more trouble than it's worth as string as such a common thing in so many languages.

IE, pretty soon the fact that your language has a few strings OVERWHELMS *every other detail* of that language, which is not a good thing.

Yes, lets just go with the default.